cinder/volume/drivers/lightos.py

```diff
@@ -1091,12 +1092,29 @@ def __add_volume_acl(self, project_name, lightos_volname, acl_to_add, host_ips):
         if acl_to_add not in acl:
            acl.append(acl_to_add)
 
-        ip_acl=[]    
+        if 'ALLOW_NONE' in ip_acl:
+            ip_acl.remove('ALLOW_NONE')
+        
         if self.use_ip_acl():
-           ip_acl=host_ips
+            ip_acl = list(set(ip_acl).union(set(host_ips)))
         else:
            ip_acl.append('ALLOW_ANY')
-                
+
+        # The max (16) elemenets are allowed in IPACL.
+        # if elements are more than 16 then remove 
+        # less-frequently used IPv6 address(s).   
+           
+        while (len(ip_acl)>16):
```

Comment:
this looks like potential bug
ipacl will be greater than 16.
try to remove - fail

and you are stuck forever

Reply:
The idea to remove ipv6 addresses is good, but I would do something like this (untested)
```
ipv4addrs = [addr in ip_acl if netutils.is_valid_ipv4(addr)]
ipv6addrs = [addr in ip_acl if netutils.is_valid_ipv6(addr)]

alladdrs = set(ipv4addrs).union(ipv6addrs)
if len(alladdrs) <= 16:
  return alladdrs
if (len(ipv4addrs) > 16:
  # log a big fat warning and use the first 16 and hope for the best
else:
  # log a big fat warning that we are dropping some ipv6 addresses
  # use all of the ipv4 addresses and as many ipv6addrs as can fit
```

Reply:
@muliby-lb @yuval-lb updated code.